zone/client.cpp

```diff
@@ -12317,7 +12334,8 @@ void Client::PlayerTradeEventLog(Trade *t, Trade *t2)
 	};
 
 	RecordPlayerEventLogWithClient(trader, PlayerEvent::TRADE, e);
-	RecordPlayerEventLogWithClient(trader2, PlayerEvent::TRADE, e);
```

Comment:
We recorded each side of a trade for each player, is there a reason we removed this ?

Reply:
I removed after looking at the data recorded.  The only difference is the details for the other side of the trade for the basic player event.  Otherwise, the json is the same.
| id | account_id | character_id | zone_id | instance_id | x | y | z | heading | event_type_id | event_type_name | event_data | etl_table_id | created_at | 
| ---: | ---: | ---: | ---: | ---: | ---: | ---: | ---: | ---: | ---: | --- | --- | ---: | --- | 
| 2118 | 24 | 84 | 151 | 106 | -600.879 | 1250.4 | -27.6012 | 40.25 | 27 | Trade | {"character_1_id":44,"character_1_name":"Rola","character_2_id":84,"character_2_name":"Grumep","character_1_give_money":{"platinum":0,"gold":0,"silver":0,"copper":0},"character_2_give_money":{"platinum":0,"gold":0,"silver":0,"copper":0},"character_1_give_items":[],"character_2_give_items":[{"slot":3000,"item_id":8657,"charges":1,"aug_1_item_id":0,"aug_2_item_id":0,"aug_3_item_id":0,"aug_4_item_id":0,"aug_5_item_id":0,"in_bag":false}]} | 0 | 2024-11-16 18:59:28 | 
| 2117 | 11 | 44 | 151 | 106 | -596.625 | 1263.38 | -28.8502 | 14.75 | 27 | Trade | {"character_1_id":44,"character_1_name":"Rola","character_2_id":84,"character_2_name":"Grumep","character_1_give_money":{"platinum":0,"gold":0,"silver":0,"copper":0},"character_2_give_money":{"platinum":0,"gold":0,"silver":0,"copper":0},"character_1_give_items":[],"character_2_give_items":[{"slot":3000,"item_id":8657,"charges":1,"aug_1_item_id":0,"aug_2_item_id":0,"aug_3_item_id":0,"aug_4_item_id":0,"aug_5_item_id":0,"in_bag":false}]} | 0 | 2024-11-16 18:59:28 |

Reply:
Then, when they parse through the etl, the etl tables end up have duplicate info in both player_event_trade and player_event_trade_entries

That is why I removed the second version.

Reply:
It was intentional for us to double record a trade because you have two sides. You might be investigating a trade a player made but the other player doesn't get associated with it if you don't record both sides. It's fine if the data is the same